clippy_lints/src/methods/filter_map_bool_then.rs

```diff
@@ -0,0 +1,51 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::paths::BOOL_THEN;
+use clippy_utils::source::snippet_opt;
+use clippy_utils::ty::is_copy;
+use clippy_utils::{is_from_proc_macro, match_def_path, peel_blocks};
+use rustc_errors::Applicability;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::{LateContext, LintContext};
+use rustc_middle::lint::in_external_macro;
+use rustc_span::Span;
+
+use super::FILTER_MAP_BOOL_THEN;
+
+pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'tcx>, arg: &Expr<'_>, call_span: Span) {
+    if !in_external_macro(cx.sess(), expr.span)
+        && let ExprKind::Closure(closure) = arg.kind
+        && let body = cx.tcx.hir().body(closure.body)
+        && let value = peel_blocks(body.value)
+        // Indexing should be fine as `filter` always has 1 input, we unfortunately need both
+        // `inputs` and `params` here as we need both the type and the span
```

Comment:
I don't think the method was ever confirmed to be `Iterator::filter_map`.

Reply:
It is, it's under ("filter_map", [arg])

Reply:
That only constrains the method name and arity. It doesn't limit it to being the one defined on `Iterator`.

Reply:
Ah I see now. Yeah I'll add an `is_trait_method` call